LibraryApp/src/app/screens/Login/index.js

```diff
@@ -20,27 +20,21 @@ class LoginContainer extends Component {
   onPasswordChange = password => this.setState({ password });
 
   handleSubmit = () => {
-    const { dispatch } = this.props.navigation;
+    const { dispatch: navigatorDispatch } = this.props.navigation;
+    const { login } = this.props;
     const { email, password } = this.state;
     const isValidEmail = isEmailValid(email);
     const isValidPassword = isPasswordValid(password);
     this.setState({
       emailError: isValidEmail ? null : EMAIL_ERROR,
       passwordError: isValidPassword ? null : PASSWORD_ERROR
     });
-    return (
-      isValidEmail &&
-      isValidPassword &&
-      dispatch(
-        NavigationActions.navigate({
-          routeName: Routes.Library
-        })
-      )
-    );
+    return isValidEmail && isValidPassword && login(email, password, navigatorDispatch);
```

Comment:
You dont need to send the navigator dispatch, you should dispatch navigate in the action